cloudferrylib/os/image/glance_image.py

```diff
@@ -320,13 +319,61 @@ def deploy(self, info):
                          dst_image.is_public)
             dst_images[image_key] = dst_image
 
-        for image_id_src, gl_image in info['images'].iteritems():
-            img = gl_image['image']
+        # Calculation to show the progress
+        num_public, num_private, num_migrated = 0, 0, 0
+        list_public, list_private, list_migrated = [], [], []
+        total_size, migrated_size = 0, 0
+        for image_id in info['images']:
+            img = info['images'][image_id]['image']
+            image_key = (img['name'], img['owner_name'], img['checksum'],
+                         img['is_public'])
+            dst_image = dst_images.get(image_key)
+            if dst_image:
+                num_migrated += 1
+                migrated_size += dst_image.size
+                list_migrated.append('%s (%s)' % (dst_image.name,
+                                                  dst_image.id))
+                continue
+
+            total_size += img['size']
+            if img['is_public']:
+                num_public += 1
+                list_public.append('%s (%s)' % (img['name'], img['id']))
+            else:
+                num_private += 1
+                list_private.append('%s (%s)' % (img['name'], img['id']))
+        num_total_images = num_private + num_public + num_migrated
+        total_size_hr = sizeof_format.sizeof_fmt(total_size)
+        migrated_size_hr = sizeof_format.sizeof_fmt(migrated_size)
+
+        LOG.info('Total number of images to be migrated: %d, '
+                 'total size: %s\n'
+                 'Number of private images: %d\n'
+                 'Number of public images: %d\n'
+                 'Number of already migrated images: %d, total size: %s',
+                 num_total_images, total_size_hr,
+                 num_private, num_public, num_migrated, migrated_size_hr)
+        LOG.info('List of private images:\n%s', '\n'.join(list_private))
+        LOG.info('List of public images:\n%s', '\n'.join(list_public))
+        LOG.info('List of migrated images:\n%s', '\n'.join(list_migrated))
```

Comment:
Let's move all that logic outside of `deploy()` method, `deploy()` is already way too complex